src/walletSdk/index.ts

```diff
@@ -1,18 +1,115 @@
+import StellarSdk, { Networks, Server } from "stellar-sdk";
+
 import { Anchor } from "./anchor";
+import { WalletSigner, DefaultSigner } from "./auth/WalletSigner";
+import { Stellar } from "./horizon/Stellar";
+import { NETWORK_URLS } from "./horizon/constants";
+import { Recovery } from "./recovery/Recovery";
+import { getUrlDomain } from "./util/url";
+
+// TODO - https://stellarorg.atlassian.net/browse/WAL-789?atlOrigin=eyJpIjoiZjcwNzZhOTJmNjE1NGRhNTk1NDlkOTExMTYxMDJkZmYiLCJwIjoiaiJ9
+interface ClientConfigFn {}
+interface ClientConfig {}
+class HttpClient {}
 
-export enum NETWORKS {
-  PUBLIC = "PUBLIC",
-  TESTNET = "TESTNET",
+export class Config {
+  app: ApplicationConfiguration;
+  stellar: StellarConfiguration;
+  constructor(stellarCfg, appCfg) {
+    this.stellar = stellarCfg;
+    this.app = appCfg;
+  }
 }
 
 export class Wallet {
-  network: NETWORKS = NETWORKS.TESTNET;
+  private cfg: Config;
+  private clients = [];
+
+  static TestNet = (): Wallet => {
+    return new Wallet(StellarConfiguration.TestNet());
+  };
+
+  static MainNet = (): Wallet => {
+    return new Wallet(StellarConfiguration.MainNet());
+  };
+
+  constructor(
+    stellarConfiguration: StellarConfiguration,
+    applicationConfiguration: ApplicationConfiguration = new ApplicationConfiguration()
+  ) {
+    this.cfg = new Config(stellarConfiguration, applicationConfiguration);
+  }
+
+  anchor(homeDomain: string, httpClientConfig: ClientConfigFn = null) {
+    const url =
+      homeDomain.indexOf("://") !== -1 ? homeDomain : `https://${homeDomain}`;
```

Comment:
With this approach there should be a configuration option in client to enable http somehow. 
In Kotlin, with one config scheme is changed to http (and I'm setting https by default in URL)
TS should has something like this too

Reply:
+1

Reply:
gotchya, I can add that as part of [this ticket ](https://stellarorg.atlassian.net/browse/WAL-789?atlOrigin=eyJpIjoiYjFkOTU5ZDFkZDkxNDZhYmJhNmNiM2U3MTYxZWQyMDQiLCJwIjoiaiJ9)when adding the httpclient stuff. I made a note of it on that ticket